ruby/vampires.rb

```diff
@@ -2,76 +2,100 @@
+	# Recieve applicant name.
+	puts "Welcome to Werewolf Inc.  Please state your name."
+	name = gets.chomp
+	# Recieve applicant stated age.
+	puts "Please state your age."
+	stated_age = gets.chomp.to_i
+	# Recieve applicant stated birthday
+	puts "Please state your year of birth."
+	birth_year = gets.chomp.to_i
+	birth_year_age = 2016 - birth_year
+	# Age honest is a boolean variable testing if state birth year corresponds with state age.
+	age_honesty = (stated_age == birth_year_age)
+	# Test applicant garlic preference. Support invalid input.
+	valid_input = false
+	until valid_input
+		puts "Would you like garlic bread with your lunch? y/n?"
+		garlic = gets.chomp
+		garlic = garlic.downcase
+		if garlic == "y"
+			garlic = true
+			valid_input = true
+		elsif garlic == "n"
+			garlic = false
+			valid_input = true
+		else
+			puts ""
+			puts "I didn't understand you."
+		end
 	end
-end
-# Assess applicant interest in health insurance. Support invalid input.
-valid_input = false
-until valid_input
-	puts "Are you interested in enrolling in our health insurance plan? y/n?"
-	insurance = gets.chomp
-	insurance.downcase
-	if insurance == "y"
-		insurance = true
-		valid_input = true
-	elsif insurance == "n"
-		insurance = false
-		valid_input = true
+	# Assess applicant interest in health insurance. Support invalid input.
+	valid_input = false
+	until valid_input
+		puts "Are you interested in enrolling in our health insurance plan? y/n?"
+		insurance = gets.chomp
+		insurance.downcase
+		if insurance == "y"
+			insurance = true
+			valid_input = true
+		elsif insurance == "n"
+			insurance = false
+			valid_input = true
+		else
+			puts ""
+			puts "I didn't understand you."
+		end
+	end
+
+	#Vampire detection program.
+	if age_honesty == true && (garlic == true || insurance == true)
+		vampire_status = "Probably not a vampire."
+		possibility_1 = true
+	elsif (age_honesty == false && (garlic == false && insurance == true)) || (age_honesty == false && (garlic == true && insurance == false))
+		vampire_status = "Probably a vampire."
+		possibility_2 = true
+	elsif (age_honesty == false && (garlic == false && insurance == false))
+		vampire_status = "Almost certainly a vampire."
+		possibility_3 = true
 	else
-		puts ""
-		puts "I didn't understand you."
+		vampire_status = "Results inconclusive."
 	end
-end
 
-#Vampire detection program.
-if age_honesty == true && (garlic == true || insurance == true)
-	vampire_status = "Probably not a vampire."
-	possibility_1 = true
-elsif (age_honesty == false && (garlic == false && insurance == true)) || (age_honesty == false && (garlic == true && insurance == false))
-	vampire_status = "Probably a vampire."
-	possibility_2 = true
-elsif (age_honesty == false && (garlic == false && insurance == false))
-	vampire_status = "Almost certainly a vampire."
-	possibility_3 = true
-else
-	vampire_status = "Results inconclusive."
+	if (possibility_1 == true) && (name == "Drake Cula" || name == "Tu Fang")
```

Comment:
I think this should be your first check. If their name is Drac or Fang then we know we have a vampire. You can nest the rest of the checks under an else following this check. if vamp == true then no reason to check anything else.

The below is an example of handling the control flow in a more efficient computational pattern. Also, it is important to practice nesting conditions as many control flows will require you to do so.

```
if name == vamp
      vampire = true
else
      if other conditions
      elsif other conditions
      end
end
```